library/thread/sizedqueue/shared/enque.rb

```diff
@@ -0,0 +1,38 @@
+describe :sizedqueue_enq, :shared => true do
+  it "blocks if queued elements exceed size" do
+    q = @object.new(1)
+
+    q.size.should == 0
+    q.send(@method, :first_element)
+    q.size.should == 1
+
+    blocked_thread = Thread.new { q.send(@method, :second_element) }
+    sleep 0.01 until blocked_thread.stop?
+
+    q.size.should == 1
+    q.pop.should == :first_element
+    q.size.should == 0
+
+    blocked_thread.join
+    q.size.should == 1
+    q.pop.should == :second_element
+    q.size.should == 0
+  end
+
+  ruby_version_is "2.2" do
+    it "raises a ThreadError if queued elements exceed size when not blocking" do
+      q = @object.new(2)
+      method = @method
```

Comment:
I did that while trying to understand some unexpected behavior that I thought may have been scope related.  Once I understood what was happening, I meant to delete it.  Sorry about that.  I will fix it.